sysprep/instance_setup.ps1

```diff
@@ -128,14 +128,33 @@ function Change-InstanceProperties {
   $nics | Invoke-CimMethod -Name SetDNSServerSearchOrder -Arguments @{DNSServerSearchOrder=$null}
   Write-Log 'All networks set to DHCP.'
 
+  # Find which interface type is being used
   $netkvm = Get-CimInstance Win32_NetworkAdapter -filter "ServiceName='netkvm'"
-  $netkvm | ForEach-Object {
-    Invoke-ExternalCommand netsh interface ipv4 set interface $_.NetConnectionID mtu=1500 | Out-Null
+  $gvnic = Get-CimInstance Win32_NetworkAdapter -filter "ServiceName='gvnic'"
+  if ($netkvm -ne $null) {
+    $interface = $netkvm
+    Write-Log 'VirtIO network adapter detected.'
   }
-  Write-Log 'MTU set to 1500.'
+  elseif ($gvnic -ne $null) {
+    $interface = $gvnic
+    Write-Log 'gVNIC network adapter detected.'
+  }
+  else {
+    Write-Log 'Error retrieving network adapter, unable to identify network adapter as gVNIC or VirtIO.'
+  }
+
+  if ($interface -ne $null) {
+    $interface | ForEach-Object {
+      Invoke-ExternalCommand netsh interface ipv4 set interface $_.NetConnectionID mtu=1500 | Out-Null
+    }
+    Write-Log 'MTU set to 1500.'
 
-  Invoke-ExternalCommand route /p add 169.254.169.254 mask 255.255.255.255 0.0.0.0 if $netkvm[0].InterfaceIndex metric 1 -ErrorAction SilentlyContinue
-  Write-Log 'Added persistent route to metadata netblock via first netkvm adapter.'
+    Invoke-ExternalCommand route /p add 169.254.169.254 mask 255.255.255.255 0.0.0.0 if $netkvm[0].InterfaceIndex metric 1 -ErrorAction SilentlyContinue
+    Write-Log 'Added persistent route to metadata netblock via first adapter.'
```

Comment:
Since you have the object you can include the adapter name instead the message instead of using the generic "via first adapter"

Reply:
Is the approach taken here better? I kept single quotes to be consistent with the way other log messages are generated, however using double quotes would allow substitution and we could send a single string back to the logger (versus using the + method). The latter approach would be inconsistent with formatting vs the other log entries, though. Wdyt?